### PR TITLE
feat(backend): Introduce OrganizationRoleAPI

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

Add OrganizationRoleAPI for CRUD operations regarding instance level organization roles.

*This will be backported*

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { joinPaths } from '../../util/path';
+import type { DeletedObject } from '../resources';
+import type { OrganizationRole } from '../resources/OrganizationRole';
+import { AbstractAPI } from './AbstractApi';
+
+const basePath = '/organizations_roles';
+
+type GetOrganizationRoleListParams = {
```
Comment (chanioxaris) [inline packages/backend/src/api/endpoints/OrganizationRoleApi.ts:8]: 🔧 We also support `query` & `order_by` parameters

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+type CreateParams = {
+  name: string;
+  key: string;
+  description: string;
+  permissions: string[];
+};
+
+type GetOrganizationRoleParams = { roleId: string };
+
+type UpdateParams = {
```
Comment (chanioxaris) [inline packages/backend/src/api/endpoints/OrganizationRoleApi.ts:22]: 🔧 Can also update the `key`

File: packages/backend/src/api/resources/JSON.ts [hunk 3]
```diff
@@ -172,6 +173,16 @@ export interface OrganizationMembershipPublicUserDataJSON {
   user_id: string;
 }
 
+export interface OrganizationRoleJSON extends ClerkResourceJSON {
+  object: ObjectType.Role;
+  name: string;
+  key: string;
+  description: string;
+  permissions: string[];
```
Comment (chanioxaris) [inline packages/backend/src/api/resources/JSON.ts:181]: 🔧 The permissions are an array of permission objects

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  name?: string;
+  description?: string;
+  //TODO(@pantelis): Will this override or union ?
+  permissions?: string[];
```
Comment (panteliselef) [inline packages/backend/src/api/endpoints/OrganizationRoleApi.ts:26]: @chanioxaris when updating permissions, is it an addition or override ?

Comment (chanioxaris) [reply packages/backend/src/api/endpoints/OrganizationRoleApi.ts:26]: It's an override. For additions, you have to use the dedicated endpoint

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts (hunk 1 above)
Comment (panteliselef) [reply packages/backend/src/api/endpoints/OrganizationRoleApi.ts:8]: @chanioxaris is order_by a string or can i provide a stricter type ?

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts (hunk 4 above)
Comment (panteliselef) [reply packages/backend/src/api/endpoints/OrganizationRoleApi.ts:26]:
```ts
{ permissions: undefined } // will not override
{ permissions: [] } // will assign no perms to a role
```
@chanioxaris is the above correct ?

**Review by dimkl — APPROVED** [review]

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts (hunk 1 above)
Comment (chanioxaris) [reply packages/backend/src/api/endpoints/OrganizationRoleApi.ts:7]: It's just a string

File: packages/backend/src/api/endpoints/OrganizationRoleApi.ts (hunk 4 above)
Comment (chanioxaris) [reply packages/backend/src/api/endpoints/OrganizationRoleApi.ts:66]: That's correct!

**Review by chanioxaris — APPROVED** [review]

**Pull request merged**